test/640_proxy_restart_reattaches_test.sh

```diff
@@ -9,11 +9,13 @@ NAME=seetwo.weave.local
 start_suite "Proxy restart reattaches networking to containers"
 
 weave_on $HOST1 launch
-proxy docker_on $HOST1 run -e WEAVE_CIDR=$C2/24 -dt --name=c2 -h $NAME $SMALL_IMAGE /bin/sh
-proxy docker_on $HOST1 run -e WEAVE_CIDR=$C1/24 -dt --name=c1          $DNS_IMAGE   /bin/sh
+proxy docker_on $HOST1 run -e WEAVE_CIDR="$C2/24 net:10.40.0.0/30" -dt --name=c2 -h $NAME $SMALL_IMAGE /bin/sh
+proxy docker_on $HOST1 run -e WEAVE_CIDR=$C1/24               -dt --name=c1          $DNS_IMAGE   /bin/sh
 
+IP2=$(weave_on $HOST1 ps $2 | grep -o -E '10\.40\.0\.[12]')
 proxy docker_on $HOST1 restart c2
 assert_raises "proxy exec_on $HOST1 c2 $CHECK_ETHWE_UP"
 assert_dns_record $HOST1 c1 $NAME $C2
+assert_dns_record $HOST1 c1 $NAME $IP2
```

Comment:
I thought it was reasonable to test the case where the container has more than one address.